### ISSUE TITLE
[no-duplicate-imports] false positives when used with 'import type'

### ISSUE THREAD

**Issue by richard-viney** [typescript-eslint/typescript-eslint#2636]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-duplicate-imports": "error"
  }
}
```

I've run into some false positives with this new rule when used with `import type`. The following code triggers this rule:

```TS
import type Foo from "my-module";
import type { Bar } from "my-module";
```

But it is not actually possible to write

```ts
import type Foo, { Bar } from "my-module";
```

because that results in the TypeScript error

```
A type-only import can specify a default import or named bindings, but not both. ts(1363)
```

This rule may need some additional logic to handle this case.

**Expected Result**

No errors reported by this rule for the given code.

**Actual Result**

The `no-duplicate-imports` rule reports a linting error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.4.0` |
| `@typescript-eslint/parser`        | `4.4.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `12.18.4` |
